### PR TITLE
Turn on caching for testing datasets and other things

### PR DESCRIPTION
**Pull request by marcellp**
# Background
#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
     "micromatch": "^4.0.2",
     "node-cleanup": "^2.1.2",
     "node-fetch": "^2.6.1",
+    "node-jq": "^2.0.0",
```
Comment (ivababukova) [inline package.json:54]: can you remove `jq-web`

File: src/api/events/index.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ module.exports = (socket) => {
       AWSXRay.capturePromise();
       AWSXRay.setSegment(segment);
 
-      logger.log('Work submitted from client', socket.id, ':', data);
+      logger.log(`[REQ ??, SOCKET ${socket.id}] Work submitted from client.`);
+      logger.log(`[REQ ??, SOCKET ${socket.id}] ${JSON.stringify(data)}`);
```
Comment (ivababukova) [inline src/api/events/index.js]: use prettify in here

File: src/api/general-services/work-submit/create-worker-k8s.js [hunk 3]
```diff
@@ -14,17 +14,17 @@ const createWorkerResources = async (service) => {
   const [assignedPods, unassignedPods] = await Promise.all(
     [
       k8sApi.listNamespacedPod(namespace, null, null, null, null, `experimentId=${experimentId}`),
-      k8sApi.listNamespacedPod(namespace, null, null, null, null, '!experimentId'),
+      k8sApi.listNamespacedPod(namespace, null, null, null, null, '!experimentId,!run'),
```
Comment (ivababukova) [inline src/api/general-services/work-submit/create-worker-k8s.js:20]: there is a comma here, is this correct?

File: src/api/general-services/work-submit/index.js [hunk 4]
```diff
@@ -116,24 +117,12 @@ class WorkSubmitService {
...
-          try {
-            const response = await this.getQueueAndHandleMessage();
-            resolve(response);
-          } catch (error) {
-            reject(error);
-          } finally {
-            subsegment.close();
-          }
-        });
-      }),
```
Comment (ivababukova) [inline src/api/general-services/work-submit/index.js:134]: why deleting all this?

File: src/cache/createClient.js [hunk 5]
```diff
@@ -38,7 +38,8 @@ const createClient = (options) => {
   });
 
   redis.on('error', (error) => {
-    logger.error(`redis:${endpoint}`, 'An error occurred:', error.message);
+    logger.error(`redis:${endpoint}`, 'An error occurred:');
+    logger.error(`redis:${endpoint}`, error);
```
Comment (ivababukova) [inline src/cache/createClient.js:42]: would that print the message as well?

File: src/cache/generate-config.js [hunk 6]
```diff
@@ -8,7 +7,7 @@ const BASE_CONFIG = {
   primary: { host: 'localhost', port: '6379' },
   reader: { host: 'localhost', port: '6379' },
   retryDelay: 10000,
-  redisGetTimeout: 3000,
+  redisGetTimeout: null,
```
Comment (ivababukova) [inline src/cache/generate-config.js:10]: why removing the timeout

File: src/config/default-config.js [hunk 7]
```diff
@@ -94,7 +94,7 @@ if (config.clusterEnv === 'staging' && config.sandboxId === 'default') {
 if (config.clusterEnv === 'staging' && config.sandboxId !== 'default') {
   config.workerInstanceConfigUrl = `https://raw.githubusercontent.com/biomage-ltd/iac/master/releases/staging/${config.sandboxId}.yaml`;
   config.pipelineInstanceConfigUrl = `https://raw.githubusercontent.com/biomage-ltd/iac/master/releases/staging/${config.sandboxId}.yaml`;
-  config.cachingEnabled = false;
+  config.cachingEnabled = true;
```
Comment (ivababukova) [inline src/config/default-config.js]: changes in this file need to be reverted

File: src/loaders/express.js [hunk 8]
```diff
@@ -122,6 +127,7 @@ module.exports = async (app) => {
       methods: ['GET', 'POST'],
       credentials: true,
     },
+    pingTimeout: 600000,
```
Comment (ivababukova) [inline src/loaders/express.js:133]: where is this `pingTimeout` used?

File: src/utils/authMiddlewares.js [hunk 9]
```diff
@@ -103,7 +101,7 @@ const authenticationMiddlewareSocketIO = async (authHeader) => {
       const { kid } = header;
 
       cache.get(kid)
-        .then((pem) => callback(null, pem))
+        .then(({ pem }) => callback(null, pem))
```
Comment (ivababukova) [inline src/utils/authMiddlewares.js:104]: you sure about these brackets?

File: src/api/general-services/work-submit/create-worker-k8s.js (hunk 3 above)
Comment (ivababukova) [reply src/api/general-services/work-submit/create-worker-k8s.js:20]: add a comment on what this does

**Review by kafkasl — APPROVED** [review]

**Pull request merged**